src/theory/quantifiers/sygus/sygus_reconstruct.cpp

```diff
@@ -318,7 +318,7 @@ TypeBuiltinSetMap SygusReconstruct::matchNewObs(Node t, Node sz)
   matches.insert(d_sygusVars.cbegin(), d_sygusVars.cend());
 
   // try to match the builtin term with the pattern sz
-  if (expr::match(rewrite(datatypes::utils::sygusToBuiltin(sz)), t, matches))
+  if (expr::match(convert(rewrite(datatypes::utils::sygusToBuiltin(sz))), convert(t), matches))
```

Comment:
Can you move this line to its own method?  And perhaps separate it out into multiple lines.

(We will want to experiment with using single invocation subcalls as a replacement).